### PR TITLE
Start new day modal

### PR DESCRIPTION
**Pull request by Stas-Buzunko**

### CODE REVIEW

File: src/components/CountDown.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react'
+
+class CountDown extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (Stas-Buzunko) [inline src/components/CountDown.js:6]: empty state, no reason for constructor

File: src/components/CountDown.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  renderTime() {
+    if (this.props.how_long) {
+      this.tillMidnight()
+    } else {
+      this.h24()
+    }
+  }
+
+  tillMidnight() {
```
Comment (Stas-Buzunko) [inline src/components/CountDown.js:27]: tillMidnight and h24 are 90% the same
you don't need 2 methods for this

File: src/components/NewActivityList.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      hours: '',
+      money: '',
+      how_long: true,
+      startedAt: '',
+      isModalShown: true
+    }
+  }
+
+  componentDidMount() {
+    const userId = firebase.auth().currentUser.uid;
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js:23]: it's not a mistake at all, but i like this more, it's up to you to decide which one looks better

const { uid }  = firebase.auth().currentUser

if you choose mine, don't forget to replace userId with uid

File: src/components/NewActivityList.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    firebase.database().ref('/users/' + userId + '/settings').once('value').then(snapshot => {
+      const currentRate = (snapshot.val().currentRate * 0.4)
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js:25]: what will be if rate is not specified in settings?
I always recommend checking if snapshot.val() !== null to make sure we have value there

Comment (Stas-Buzunko) [reply src/components/NewActivityList.js:25]: how about fetching user and saving to state in App.js and then passing to Child routes?

File: src/components/NewActivityList.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      this.setState({
+        [field]: value,
+      })
+    }
+  }
+
+  addChallenge = () => {
+    const { hours, money, how_long } = this.state
+    if ((hours.length > 0) && (money > 0)) {
+      this.setState({error: ''});
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js:54]: how about moving error: '' to challengeSave?
this way you remove extra setState call which is expensive

File: src/components/NewActivityList.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+      this.challengeSave({hours, money, how_long});
+    } else {
+      toastr.error('Please enter correct details');
+    }
+  }
+
+  challengeSave(params) {
+    const { hours, money, how_long } = params
+    const userUid = this.props.user.uid
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js:63]: I'd prefer

 const { uid } = this.props.user

but it's not a mistake

File: src/components/NewActivityList.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    const key = firebase.database().ref('days').push().key
+    firebase.database().ref('days/' + key).update({
+      hours,
+      money,
+      how_long,
+      userUid,
+      startedAt: Date.now(),
+      isDayOver: false,
+      isCharged: false
+    }).then(() => toastr.success('Your Activity saved!'))
```
Comment (Stas-Buzunko) [inline src/components/NewActivityList.js:73]: I'd prefer to have
.then() on new line
but it's not a mistake, just a bit more readable in my opinion

File: src/components/CountDown.js (hunk 1 above)
Comment (Xirisss) [reply src/components/CountDown.js:6]: state тут юзается ниже. Если убрать конструктор, то как бы я могу вызывать props без super(props)?

File: src/components/NewActivityList.js (hunk 3 above)
Comment (Xirisss) [inline src/components/NewActivityList.js]: done

File: src/components/CountDown.js (hunk 2 above)
Comment (Xirisss) [inline src/components/CountDown.js]: done

File: src/components/NewActivityList.js (hunk 5 above)
Comment (Xirisss) [reply src/components/NewActivityList.js:54]: done

File: src/components/NewActivityList.js (hunk 4 above)
Comment (Xirisss) [reply src/components/NewActivityList.js:25]: 1.done

File: src/components/NewActivityList.js (hunk 3 above)
Comment (Xirisss) [reply src/components/NewActivityList.js:23]: done

File: src/components/CountDown.js (hunk 2 above)
Comment (Xirisss) [reply src/components/CountDown.js:27]: done

File: src/components/NewActivityList.js (hunk 6 above)
Comment (Xirisss) [reply src/components/NewActivityList.js:63]: done

File: src/components/NewActivityList.js (hunk 7 above)
Comment (Xirisss) [reply src/components/NewActivityList.js:73]: done

File: src/components/NewActivityList.js (hunk 4 above)
Comment (Xirisss) [reply src/components/NewActivityList.js:25]: 2.done

**Pull request closed without merging**